### PR TITLE
Add tests examples

### PR DESCRIPTION
**Pull request by SergeyBerdnikovich**

### CODE REVIEW

File: ginkgo_me/BBQ_gotest_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	coalFarmMock    *mock_coalfarm.MockProvider
+
+	subject Provider
+)
+
+func TestCoolPriceCalculator2(t *testing.T) {
+	t.Run("#CalculatePrice", func(t *testing.T) {
+		var (
+			numberOfPeople int
+			totalPrice     int
```
Comment (artiomgiza) [inline ginkgo_me/BBQ_gotest_test.go:29]: why do we need this definition here?
you might use `:=` in line 52 (and others) and remove this from here. 
wdyt?

Comment (artiomgiza) [reply ginkgo_me/BBQ_gotest_test.go:29]: same for others

**Review by artiomgiza — COMMENTED** [review]
I think there is a good approach to write tests in the same structure as `function under test` flow.
When test cases are written in different way - it's very hard to understand what was tested.
WDYT?

File: ginkgo_me/BBQ.go [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 
 import (
-	beeffarm "test_me/pkgs/beef-farm"
-	chickenfarm "test_me/pkgs/chicken-farm"
-	coalfarm "test_me/pkgs/coal-farm"
-	mangalstore "test_me/pkgs/mangal-store"
+	beeffarm "github.com/artiomgiza/test_me/pkgs/beef-farm"
+	chickenfarm "github.com/artiomgiza/test_me/pkgs/chicken-farm"
+	coalfarm "github.com/artiomgiza/test_me/pkgs/coal-farm"
+	mangalstore "github.com/artiomgiza/test_me/pkgs/mangal-store"
```
Comment (artiomgiza) [inline ginkgo_me/BBQ.go:7]: do you know why this is like this?
looks like your is righter, interesting why mine is relative...

File: pkgs/beef-farm/mock/beef_farm_mock.go [hunk 3]
```diff
@@ -5,8 +5,9 @@
 package mock_beeffarm
 
 import (
```
Comment (artiomgiza) [inline pkgs/beef-farm/mock/beef_farm_mock.go:7]: why did you regenerate mocks? how it's related to this PR?

File: ginkgo_me/BBQ_gotest_test.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
+package ginkgo_me
```
Comment (artiomgiza) [inline ginkgo_me/BBQ_gotest_test.go:1]: plz add your examples with Readme explanation to different dir (with appropriative maming to make it's clear what this example shows)

File: ginkgo_me/BBQ_ginkgo_test.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+					})
+				})
+
+				Context("when chickenFarm is failed", func() {
+					BeforeEach(func() {
+						chickenFarmMock.EXPECT().GetPullet(numberOfPeople).Return(0, errors.New("Error"))
+					})
+
+					It("returns an error", func() {
+						Expect(err).NotTo(BeNil())
```
Comment (artiomgiza) [inline ginkgo_me/BBQ_ginkgo_test.go:112]: FYI, this might be replaced by:
`Expect(err).To(Succeed())`

File: ginkgo_me/BBQ.go (hunk 2 above)
Comment (SergeyBerdnikovich) [reply ginkgo_me/BBQ.go:7]: your way didn't work for me :(

File: pkgs/beef-farm/mock/beef_farm_mock.go (hunk 3 above)
Comment (SergeyBerdnikovich) [reply pkgs/beef-farm/mock/beef_farm_mock.go:7]: because they didn't work for me :(

Comment (artiomgiza) [reply pkgs/beef-farm/mock/beef_farm_mock.go:7]: Hm, ok

**Comment by artiomgiza** [comment]
@SergeyBerdnikovich tell me when to re review plz

**Pull request merged**